src/deviceAPI/AuthorizedDeviceAPI.js

```diff
@@ -0,0 +1,72 @@
+const crypto = require('crypto')
+const fs = require('fs')
+const path = require('path')
+
+const DeviceAPI = require('./DeviceAPI')
+
+/**
+* Stores information required for the secure authentication of the device
+* @param {Object}  {deviceId: **** ,pathToCertificate: ****}
+* @memberof AuthorizedDeviceAPI
+*/
+class AuthorizedDeviceAPI extends DeviceAPI {
+
+  static get HASHLENGTH() {
+    return 44
+  }
+
+  static _createHashFromCertificateFile(pathToCertificate) {
+    const certificate = fs.readFileSync(path.resolve(pathToCertificate), { encoding: 'utf-8' }).trim()
+    const hash = crypto.createHash('sha256').update(certificate).digest('base64')
+    return hash
+  }
+
+  static _validateCertificate(pathToCert) {
+
+    if (!pathToCert) {
+      throw Error('No Certificate specified')
+    }
+    try {
+      fs.readFileSync(path.resolve(pathToCert))
```

Comment:
You don't really use the file here, so `readFile` might be wrong API call. How about `existsSync`: 
https://flaviocopes.com/how-to-check-if-file-exists-node/